src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -435,15 +452,15 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
 
     // TODO: Does it really make sense to pass in the status from our last "get the IP address" operation?
     sdCtx->callback(sdCtx->context, &service, Span<Inet::IPAddress>(sdCtx->addresses.data(), sdCtx->addresses.size()), status);
-    MdnsContexts::GetInstance().Remove(sdCtx);
+    MdnsContexts::GetInstance().Removes(ContextType::GetAddrInfo);
+    MdnsContexts::GetInstance().Removes(ContextType::GetAddrInfoConnection);
```

Comment:
So if we start a lookup, then start another lookup (which shares the connection, right?), how does this removal of the context work?  When does it happen, in the lifetimes of the two lookups?

@vivien-apple

Reply:
Ok. I did change the code - it ends up beeing a bigger change that what I was expecting.

Now the code is creating a "shared" connection on a per-resolve basis. I have also updated the code that close the context to make it clearer what is the lifetime of contexts in general. This is the "GenericContext::Finalise" method.